runtime/src/main/scala/akka/grpc/scaladsl/scaladsl.scala

```diff
@@ -1,35 +1,5 @@
 package akka.grpc
 
-import java.util.Base64
-
-import akka.http.scaladsl.model.HttpHeader
-import akka.util.ByteString
-
-import scala.collection.immutable
-
 package object scaladsl {
-  type MetadataMap = immutable.Map[String, Seq[MetadataEntry]]
-
-  object MetadataMap {
-    def apply(headers: immutable.Seq[HttpHeader] = immutable.Seq.empty): MetadataMap = {
-      // REVIEWER NOTE: modeled after akka.grpc.internal.MetadataImpl.metadataMapFromGoogleGrpcMetadata
-      var entries = Map.empty[String, List[MetadataEntry]]
-      headers.foreach { header =>
-        val key = header.name()
-        val entry =
-          if (key.endsWith("-bin")) {
-            val bytes = Base64.getDecoder.decode(header.value())
-            BytesEntry(ByteString(bytes))
-          } else {
-            val text = header.value
-            StringEntry(text)
-          }
-        if (entries.contains(key)) {
-          entries += (key -> (entry :: entries(key)))
-        } else
-          entries += (key -> (entry :: Nil))
-      }
-      entries
-    }
-  }
+  type MetadataMap = Map[String, List[MetadataEntry]]
```

Comment:
I don't think we should have a type alias as public API, and the signature isn't that messy, so let's just use that instead.

Reply:
Fixed